simtools/production_configuration/calculate_statistical_errors_grid_point.py

```diff
@@ -0,0 +1,458 @@
+"""
+Provides functionality to evaluate statistical errors from dl2_mc_events_file files.
+
+Classes
+-------
+StatisticalErrorEvaluator
+    Handles error calculation for given dl2_mc_events_file files and specified metrics.
+
+
+"""
+
+import logging
+
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+
+_logger = logging.getLogger(__name__)
+
+
+class StatisticalErrorEvaluator:
+    """
+    Evaluates statistical errors from a dl2_mc_events_file file.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the dl2_mc_events_file file.
+    file_type : str
+        Type of the file, either 'point-like' or 'cone'.
+    metrics : dict, optional
+        Dictionary of metrics to evaluate. Default is None.
+    grid_point : tuple, optional
+        Tuple specifying the grid point (energy, azimuth, zenith, NSB, offset).
+    """
+
+    def __init__(
+        self,
+        file_path: str,
+        file_type: str,
+        metrics: dict[str, float],
+        grid_point: tuple[float, float, float, float, float] | None = None,
+    ):
+        """
+        Init the evaluator with a dl2_mc_events_file file, its type, and metrics to calculate.
+
+        Parameters
+        ----------
+        file_path : str
+            The path to the dl2_mc_events_file file.
+        file_type : str
+            The type of the file ('point-like' or 'cone').
+        metrics : dict, optional
+            Dictionary specifying which metrics to compute and their reference values.
+        grid_point : tuple, optional
+            Tuple specifying the grid point (energy, azimuth, zenith, NSB, offset).
+        """
+        self.file_path = file_path
+        self.file_type = file_type
+        self.metrics = metrics
+        self.grid_point = grid_point
+
+        self.data = self.load_data_from_file()
+
+        self.uncertainty_effective_area = None
+        self.error_energy_estimate_bdt_reg_tree = None
```

Comment:
We said we rename this so that the energy reconstruction method is not part of the name. That energies are reconstructed using BDT trees is not relevant in this context.

Name it `energy_estimate` (here and throughout the code)